db/db_impl/db_impl.cc

```diff
@@ -3397,6 +3397,31 @@ void DBImpl::GetApproximateMemTableStats(ColumnFamilyHandle* column_family,
 Status DBImpl::GetApproximateSizes(const SizeApproximationOptions& options,
                                    ColumnFamilyHandle* column_family,
                                    const Range* range, int n, uint64_t* sizes) {
+  const Comparator* const ucmp = column_family->GetComparator();
+  assert(ucmp);
+  size_t ts_sz = ucmp->timestamp_size();
+  if (ts_sz == 0) {
+    return GetApproximateSizesInternal(options, column_family, range, n, sizes);
+  }
+
+  std::string start_str[n];
```

Comment:
I am surprised that this line can compile now since `n` is not const...

Reply:
Though the strings themselves (may) reside on the heap, the `string` object containing pointers to the string are allocated on the stack. The size of available space on the stack for the thread is limited and can be small compared to the heap size. Therefore, if the user input `n` to this API is very large, there is a chance that `start_str`, `limit_str` and `range_with_ts` will overflow. How about just allocating from heap?